src/main/java/app/tozzi/JPASearchFunctions.java

```diff
@@ -1,71 +1,109 @@
+        = (cb, values) -> cb.like(values[0], cb.concat(cb.concat("%", values[1]), "%"));
+    public static final JPAFuncWithExpressions<String, Boolean> CONTAINS_IGNORECASE
+        = (cb, values) -> cb.like(cb.upper(values[0]), cb.concat("%", cb.concat(cb.upper(values[1]), "%")));
+    public static final JPAFuncWithExpressions<?, Boolean> NOTEQ
+        = (cb, values) -> cb.notEqual(values[0], values[1]);
+    public static final JPAFuncWithExpressions<String, Boolean> NOTEQ_IGNORECASE
+        = (cb, values) -> cb.notEqual(cb.upper(values[0]), cb.upper(values[1]));
+    public static final JPAFuncWithExpressions<Comparable, Boolean> GT
+        = (cb, values) -> cb.greaterThan(values[0], values[1]);
+    public static final JPAFuncWithExpressions<Comparable, Boolean> GTE
+        = (cb, values) -> cb.greaterThanOrEqualTo(values[0], values[1]);
+    public static final JPAFuncWithExpressions<Comparable, Boolean> LT
+        = (cb, values) -> cb.lessThan(values[0], values[1]);
+    public static final JPAFuncWithExpressions<Comparable, Boolean> LTE
+        = (cb, values) -> cb.lessThanOrEqualTo(values[0], values[1]);
+    public static final JPAFuncWithExpressions<Collection, Boolean> IN = (cb, values) -> {
+        CriteriaBuilder.In<Object> in = cb.in(values[0]);
+        var size = values.length;
+        for (var i = 1; i < size; i++) {
+            in.value(values[i]);
         }
-
         return in;
     };
-    public static final Function<FieldRootBuilderBean<?>, Predicate> NIN = s -> s.criteriaBuilder.not(IN.apply(s));
-    public static final Function<FieldRootBuilderBean<?>, Predicate> NOT_NULL = s -> s.criteriaBuilder.isNotNull(getPath(s.root, s.field));
-    public static final Function<FieldRootBuilderBean<?>, Predicate> NOT_EMPTY = s -> s.criteriaBuilder.isNotEmpty(getPath(s.root, s.field));
-    public static final Function<FieldRootBuilderBean<?>, Predicate> NULL = s -> s.criteriaBuilder.isNull(getPath(s.root, s.field));
-    public static final Function<FieldRootBuilderBean<?>, Predicate> EMPTY = s -> s.criteriaBuilder.isEmpty(getPath(s.root, s.field));
-    public static final Function<FieldRootBuilderBean<?>, Predicate> BETWEEN = s -> {
-        Collection<?> coll = (Collection<?>) s.value;
-        Iterator<?> iterator = coll.iterator();
-        return s.trim ? s.criteriaBuilder.between(s.criteriaBuilder.trim(getPath(s.root, s.field).as(String.class)), iterator.next().toString(), iterator.next().toString()) : s.criteriaBuilder.between(getPath(s.root, s.field), (Comparable) iterator.next(), (Comparable) iterator.next());
+    public static final JPAFuncWithExpressions<Collection, Boolean> NIN
+        = (cb, values) -> cb.not(IN.apply(cb, values));
+    public static final JPAFuncWithExpressions<?, Boolean> NOT_NULL
+        = (cb, values) -> cb.isNotNull(values[0]);
+    public static final JPAFuncWithExpressions<Collection, Boolean> NOT_EMPTY
+        = (cb, values) -> cb.isNotEmpty(values[0]);
```

Comment:
Can be replaced in search expressions with `["not", [...]]`

Reply:
Removed